src/Drush/Generators/AdminReadmeGridGenerator.php

```diff
@@ -0,0 +1,336 @@
+    'name' => 'PHPUnit',
+    'branches' => [
+      '4.x',
+    ],
+    'cron' => '0 6 * * *',
+    'test' => 'running-tests',
+    'checkout' => 'actions/checkout@v4',
+    'run' => 'tripal/test-tripal-action@v1.7',
+  ];
+
+  /**
+   * The workflow template file.
+   *
+   * @var string
+   */
+  private const WORKFLOW_TEMPLATE = 'readme-grid-workflow.twig';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generate(array &$vars, Assets $assets): void {
+
+    $ir = $this->createInterviewer($vars);
+    $machine_name = $ir->askMachineName();
+
+    // Must have this key.
+    $vars['machine_name'] = $machine_name;
+
+    $module = \Drupal::service('module_handler')
+      ->getModule($machine_name);
+    if (!$module) {
+      throw new \Exception('Module does not exist.');
+    }
+
+    $phpunit_yml = '%s' . DIRECTORY_SEPARATOR . self::WORKFLOW_DIR . DIRECTORY_SEPARATOR . self::WORKFLOW_FILE;
+
+    $module_path = $module->getPath();
+    $module_phpunit = sprintf($phpunit_yml, $module_path);
+
+    $is_package = FALSE;
+
+    if (!file_exists($module_phpunit)) {
+      // No workflow file asset in the module directory. Move one directory
+      // level up as it may likely be a package module.
+      $module_path = dirname($module_path, $one_level_up = 1);
+      $module_phpunit = sprintf($phpunit_yml, $module_path);
+
+      if (!file_exists($module_phpunit)) {
+        throw new \Exception('Failed to load the phpunit YML file of the module.');
+      }
+
+      $is_package = TRUE;
+    }
+
+    $apply_module = [];
+    $apply_module[] = $module->getName();
+
+    if ($is_package) {
+      // Confirm that it is a package.
+      $confirm_is_package = $ir->confirm('The module is a package, and the generator located the GitHub Workflow in the parent directory: ' . $module_path . ' and not in ' . $module->getPath() . '. Is the module a package?', TRUE);
+
+      if (!$confirm_is_package) {
+        throw new \Exception('Could not find the workflow directory. Ensure that you have setup the directory .github/workflows/ in the module path and retry the command.');
+      }
+
+      $sub_modules = [];
+
+      foreach (scandir($module_path) as $sub_module) {
+        $sub_dir = $module_path . DIRECTORY_SEPARATOR . $sub_module;
+
+        // Exclude the module name entered in the beginning prompts.
+        if (is_dir($sub_dir) && !in_array($sub_module, ['.', '..', $module->getName()])) {
+
+          // Only directory with .info.yml (a module).
+          foreach (scandir($sub_dir) as $file) {
+            if (str_contains($file, '.info.yml')) {
+              array_push($sub_modules, $sub_module);
+              break;
+            }
+          }
+        }
+      }
+
+      // Prompt to ask which sub-modules the workflow apply.
+      $apply_all = $ir->confirm('The package module (Repository) has sub-modules. Apply workflow to all [' . implode(', ', $sub_modules) . '] (Yes) or select from list (No)', TRUE);
```

Comment:
While the user/developer shouldn't need to confirm the original module in the loop below, it should be in the list displayed in this prompt.

```suggestion
      $apply_all = $ir->confirm('The package module (Repository) has sub-modules. Apply workflow to all [' . implode(', ', array_merge([$machine_name], $sub_modules)) . '] (Yes) or select from list (No)', TRUE);
```